projects/ngx-maplibre-gl/src/lib/control/fullscreen-control.directive.ts

```diff
@@ -1,48 +1,49 @@
-import {
-  AfterContentInit,
-  Directive,
-  Host,
-  HostListener,
-  Input,
-} from '@angular/core';
+import { Directive, afterNextRender, inject, input } from '@angular/core';
 import { FullscreenControl } from 'maplibre-gl';
 import { MapService } from '../map/map.service';
 import { ControlComponent } from './control.component';
 
 /**
  * `mglFullscreen` - a fullscreen control directive
- * 
+ *
  * @category Directives
  */
 @Directive({
   selector: '[mglFullscreen]',
   standalone: true,
+  host: {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    '(window:webkitfullscreenchange)': 'onFullscreen()',
+  },
 })
-export class FullscreenControlDirective implements AfterContentInit {
-  /* Init inputs */
-  @Input() container?: HTMLElement;
-  @HostListener('window:webkitfullscreenchange', ['$event.target'])
-  onFullscreen() {
-    this.mapService.mapInstance.resize();
-  }
+export class FullscreenControlDirective {
+  /* Init injection */
+  private readonly mapService = inject(MapService);
+  private readonly controlComponent = inject<
+    ControlComponent<FullscreenControl>
+  >(ControlComponent, { host: true });
 
-  constructor(
-    private mapService: MapService,
-    @Host() private controlComponent: ControlComponent<FullscreenControl>
-  ) {}
+  /* Init inputs */
+  readonly container = input<HTMLElement>();
 
-  ngAfterContentInit() {
-    this.mapService.mapCreated$.subscribe(() => {
-      if (this.controlComponent.control) {
-        throw new Error('Another control is already set for this control');
-      }
-      this.controlComponent.control = new FullscreenControl({
-        container: this.container,
+  constructor() {
+    afterNextRender(() => {
+      this.mapService.mapCreated$.subscribe(() => {
+        if (this.controlComponent.control) {
+          throw new Error('Another control is already set for this control');
+        }
+        this.controlComponent.control = new FullscreenControl({
+          container: this.container(),
+        });
+        this.mapService.addControl(
+          this.controlComponent.control,
+          this.controlComponent.position()
+        );
       });
-      this.mapService.addControl(
-        this.controlComponent.control,
-        this.controlComponent.position
-      );
     });
   }
+
+  onFullscreen() {
```

Comment:
Is this really needed? I think maplibre resize should work now without any hacks...?

Reply:
honestly I don't know, I prefer to keep it like this